R/lockfile.R

```diff
@@ -174,17 +174,27 @@ renv_lockfile_create <- function(project,
 
   lockfile <- renv_lockfile_init(project)
 
-  renv_lockfile_records(lockfile) <-
-
-    renv_snapshot_libpaths(libpaths = libpaths,
-                           project  = project) %>%
-
-    renv_snapshot_filter(project  = project,
-                         type     = type,
-                         packages = packages,
-                         exclude  = exclude) %>%
+  repeat {
+    withRestarts(
```

Comment:
I don't love the use of restarts, but this seemed like the easiest way to recompute the lockfile records, given the distance in the call sites.